pinot-common/src/main/java/org/apache/pinot/common/utils/config/InstanceUtils.java

```diff
@@ -59,6 +60,32 @@ public static String getHelixInstanceId(Instance instance) {
     return prefix + instance.getHost() + "_" + instance.getPort();
   }
 
+  public static String getInstanceAdminEndpoint(InstanceConfig instanceConfig, int defaultPort) {
+    // Backward-compatible with legacy hostname of format 'Server_<hostname>'
+    String hostname = instanceConfig.getHostName();
+    if (hostname.startsWith(Helix.PREFIX_OF_SERVER_INSTANCE)) {
+      hostname = hostname.substring(Helix.SERVER_INSTANCE_PREFIX_LENGTH);
+    }
+    return getInstanceAdminEndpoint(instanceConfig, CommonConstants.HTTP_PROTOCOL, hostname, defaultPort);
+  }
+
+  public static String getInstanceAdminEndpoint(InstanceConfig instanceConfig, String defaultProtocol, String hostname,
```

Comment:
as in ServerInstance.java, `_hostname` is precalculatd, and it can come from different fields in InstanceConfig. So I kept this method to take in that precalculated hostname directly.